Client/ViewModels/JoinGroupPageViewModel.cs

```diff
@@ -69,48 +74,74 @@ public JoinGroupPageViewModel()
             JoinGroupButtonCommand = new RelayCommand(SendJoinGroupMessage);
             CancelButtonCommand = new RelayCommand(NavigateToMain);
 
+            //Create empty group list
             GroupList = new ObservableCollection<Group>();
 
+            // Get the groups
             CommunicationHandler.ObtainedRequestedGroups += OnObtainedRequestedGroups;
             CommunicationHandler.SendRequestGroupsMessages();
```

Comment:
This is maybe a better comment:
Subscribe on event, this event will be raised when the requested groups are obtained. Then send the actual message which request the groups from the server.